web3/contract.py

```diff
@@ -778,14 +778,22 @@ def call_contract_function(contract,
     except DecodingError as e:
         # Provide a more helpful error message than the one provided by
         # eth-abi-utils
-        msg = (
-            "Could not decode contract function call {} return data {} for "
-            "output_types {}".format(
-                function_name,
-                return_data,
-                output_types
+        empty_strings = ["0x", b"0x", ""]
+        if (return_data in empty_strings
+                and contract.web3.eth.getCode(contract.address) in empty_strings):
```

Comment:
*nitpick*

I have a somewhat gigantic aversion to multi-line `if` statements.  Can you reformat this to something like the following or an alternative that doesn't split the `if` statement lines?

```python
is_missing_code_error = (
    return_data in empty_strings and
    contract.web3.eth.getCode(contract_address) in empty_strings
)
if is_missing_code_error:
    ...
```